src/layout/panel/workspaceList.js

```diff
@@ -553,22 +561,53 @@ var WorkspaceButton = GObject.registerClass(
         }
 
         handleDragOver(source, actor, x, y) {
-            if (!(source instanceof WorkspaceButton) || !this.draggable) {
-                return DND.DragMotionResult.NO_DROP;
+            if ((dragData.current.type === DRAG_TYPES.workspaceItem) && this.draggable) {
+                this.emit('drag-over', y < this.height / 2);
+                return DND.DragMotionResult.MOVE_DROP;
             }
-
-            this.emit('drag-over', y < this.height / 2);
-            return DND.DragMotionResult.MOVE_DROP;
+            if ((dragData.current.type === DRAG_TYPES.taskItem)) {
+                return DND.DragMotionResult.MOVE_DROP;
+            }
+            return DND.DragMotionResult.NO_DROP;
         }
 
-        acceptDrop(source) {
-            if (!(source instanceof WorkspaceButton)) {
-                return false;
+        acceptDrop() {
+            if (dragData.current.type === DRAG_TYPES.workspaceItem) {
+                this.emit('drag-dropped');
+                return true;
+            }
+            if (dragData.current.type === DRAG_TYPES.taskItem) {
+                const dragItem = dragData.current.item;
```

Comment:
the current item could be retrieve with the source parameter and therefore we don't need the global dragdata item